routes/api.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+use Illuminate\Support\Facades\Route;
+
+Route::post('/{location:slug}/checkout', [Tipoff\Booking\Http\Controllers\CheckoutController::class, 'store']);
+Route::get('/locations/{location:slug}', [Tipoff\Booking\Http\Controllers\LocationsController::class, 'show']);
+Route::post('/{location:slug}/contact-details', [Tipoff\Booking\Http\Controllers\ContactDetailsController::class, 'store']);
+Route::post('/{user:id}/voucher-applications', [Tipoff\Booking\Http\Controllers\VoucherApplicationsController::class, 'store']);
```

Comment:
Is `:id` required? As far I know resolver would use https://github.com/illuminate/database/blob/master/Eloquent/Model.php#L1706 by default so such syntax should be used when model is resolved by a non-default key.

Reply:
Correct, this is just copied and pasted from the existing code so far, so I haven't changed anything. But good catch.